src/hooks/screen/useMessageHook.ts

```diff
@@ -0,0 +1,73 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import {Alert, NativeSyntheticEvent} from 'react-native';
+import {ContextMenuOnPressNativeEvent} from 'react-native-context-menu-view';
+import {useCallback, useContext} from 'react';
+
+import ChatSchema from '../../database/schema/ChatSchema';
+import ShareUtils from '../../utils/share';
+import UseMessageHook from '../../../types/hooks/screens/useMessageHook.types';
+import useLocalDatabaseHook from '../../database/hooks/useLocalDatabaseHook';
+import {Context} from '../../context';
+import {setReplyTarget} from '../../context/actions/chat';
+
+function useMessageHook(): UseMessageHook {
+  const {localDb, refresh} = useLocalDatabaseHook();
+  const [replyPreview, dispatch] = (useContext(Context) as unknown as any).chat;
+
+  const setReplyPreview = useCallback((messageItem) => {
+    console.log('Reply Message');
+    setReplyTarget(messageItem, dispatch);
+  }, []);
+
+  const clearReplyPreview = useCallback(() => {
+    console.log('clear Reply target');
+    setReplyTarget(null, dispatch);
+  }, []);
+
+  const deleteMessage = (messageId: string) => {
+    if (!localDb) return;
+    try {
+      ChatSchema.updateDeletedChatType(localDb, messageId);
+      refresh('chat');
+    } catch (error) {
+      console.log('error on delete message:', error);
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Error handling could be improved here. Currently, if an error occurs during message deletion, it's only logged to the console. Consider adding user-facing error handling, such as displaying an error message to the user.

```diff
-       console.log('error on delete message:', error);
+       Alert.alert('Error', 'An error occurred while deleting the message.');
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->